nifi-extension-bundles/nifi-network-bundle/nifi-network-processors/src/main/java/org/apache/nifi/processors/network/SplitPCAP.java

```diff
@@ -0,0 +1,223 @@
+            .name("failure")
+            .description("If a FlowFile cannot be transformed from the configured input format to the configured output format, "
+            + "the unchanged FlowFile will be routed to this relationship.")
+            .build();
+    public static final Relationship REL_SPLIT = new Relationship.Builder()
+            .name("split")
+            .description("The individual 'segments' of the original FlowFile will be routed to this relationship.")
+            .build();
+
+    private static final List<PropertyDescriptor> DESCRIPTORS = List.of(PCAP_MAX_SIZE);
+
+    private static final Set<Relationship> RELATIONSHIPS = Set.of(REL_ORIGINAL, REL_FAILURE, REL_SPLIT);
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return DESCRIPTORS;
+    }
+
+    /**
+     * This method is called when a trigger event occurs in the processor.
+     * It processes the incoming flow file, splits it into smaller pcap files based on the maximum size,
+     * and transfers the split pcap files to the success relationship.
+     * If the flow file is empty or not parseable, it is transferred to the failure relationship.
+     *
+     * @param context  the process context
+     * @param session  the process session
+     */
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        int pcapMaxSize = context.getProperty(PCAP_MAX_SIZE.getName()).asInteger();
+
+        final ByteArrayOutputStream contentBytes = new ByteArrayOutputStream();
+        session.exportTo(flowFile, contentBytes);
+        final byte[] contentByteArray = contentBytes.toByteArray();
+
+        if(contentByteArray.length == 0){
+            session.putAttribute(flowFile, ERROR_REASON, "PCAP file empty.");
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        final PCAP parsedPcap;
+        final PCAP templatePcap;
+
+        // Parse the pcap file and create a template pcap object to borrow the header from.
+        try {
+            parsedPcap = new PCAP(new ByteBufferInterface(contentByteArray));
+
+            // Recreating rather than using deepcopy as recreating is more efficient in this case.
+            templatePcap = new PCAP(new ByteBufferInterface(contentByteArray));
+
+        } catch (Exception e){
+            session.putAttribute(flowFile, ERROR_REASON, "PCAP file not parseable.");
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        final List<Packet> unprocessedPackets = parsedPcap.packets();
+
+        int currentPacketCollectionSize = 0;
+        int totalFlowfileCount = 1;
+        int packetHeaderLength = 24;
```

Comment:
@JackHintonSmartDCSIT Can you please give me insight whether this is fixed or can be configurable?

Reply:
The header length is fixed, although that variable name is a bit of a misnomer as the 24-byte length header actually belongs to a PCAP file as whole rather than the individual packets (which have a 16 byte length header). I should rejig the maths here, thinking about it.

The sizes of the various headers can be found [here](https://www.tcpdump.org/manpages/pcap-savefile.5.html)